subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformBuildOperationIntegrationTest.groovy

```diff
@@ -741,4 +867,43 @@ class ArtifactTransformBuildOperationIntegrationTest extends AbstractIntegration
             acc + (acc.endsWith(":") && path.startsWith(":") ? path.substring(1) : path)
         }
     }
+
+    boolean matchTransformationIdentity(actual, TransformationIdentityWithoutId expected) {
+        actual.nodeType.toString() == ARTIFACT_TRANSFORM &&
+            actual.buildPath == expected.buildPath &&
+            actual.projectPath == expected.projectPath &&
+            actual.componentId == expected.componentId &&
+            actual.targetAttributes == expected.targetAttributes &&
+            actual.capabilities == expected.capabilities &&
+            actual.artifactName == expected.artifactName &&
+            actual.dependenciesConfigurationIdentity == expected.dependenciesConfigurationIdentity
+    }
+
+    def getPlannedNodes(int transformNodeCount) {
+        def plannedNodes = buildOperations.only(CalculateTaskGraphBuildOperationType).result.executionPlan as List<PlannedNode>
+        assert plannedNodes.every { KNOWN_NODE_TYPES.contains(it.nodeIdentity.nodeType) }
+        assert plannedNodes.count { it.nodeIdentity.nodeType.toString() == ARTIFACT_TRANSFORM } == transformNodeCount
+        return plannedNodes
+    }
+
+    def getExecuteTransformOperations(int expectOperationsCount, boolean expectFailure = false) {
+        def operations = buildOperations.all(ExecuteScheduledTransformationStepBuildOperationType)
+        assert operations.every { (it.failure != null) == expectFailure }
+        assert operations.size() == expectOperationsCount
+        return operations
+    }
+
+    void checkExecuteTransformOperation(List<BuildOperationRecord> executeOperations, TransformationIdentityWithoutId expectedTransformId, Map<String, Object> expectedDetails) {
+        def matchedOperations = executeOperations.findAll { matchTransformationIdentity(it.details.transformationIdentity, expectedTransformId) }
+        assert matchedOperations.size() == 1
+        def operation = matchedOperations[0]
+
+        with(operation.details) {
```

Comment:
🤔 I suppose you could use Spock's `verifyAll` here, WDYT?

Reply:
I didn't know about this helper function 👍

Reply:
Done